README.md

```diff
@@ -1,7 +1,42 @@
+```
+npm i @rsksmart/rif-marketplace-storage-pinning
+```
+
+## PinningManager contract
+
+### Message
+The `StorageProvider` can emit an arbitrary message in the function calls `setStorageOffer` and `emitMessage`. 
+The message is is initially intended to allow the `StorageProvider` to communicate his `nodeID`, but may be used for other purposes later.
 
 The client should be instructed to enforce the following semantics:
-- If `message` starts with `0x01`, then the following bits contain an arbitrary length `nodeID`.
\ No newline at end of file
+- If `message` starts with `0x01`, then the following bits contain an arbitrary length `nodeID`.
+
+## Glossary
+
+ - Actors
+    - **Consumer** - actor that wants his files to be persisted in the network
+    - **Provider** - actor that offers his storage for Consumers to be used in monetary exchange
+ - Entities
+    - **Offer** - Offer created by Provider that announces availability of his storage space
+      - *Billing Periods* - Specifies periods that are available to used for billing   
+      - *Billing Prices* - Specifies prices for each Billing Period, resulting in period-price tuple.
+      - *Maximum Duration* - Duration which Consumer can maximally rent the storage space for 
+      - *Total Capacity* - Total capacity that Provider is offering
+      - *Free Capacity* - Capacity in given time that is left free for Consumers to use  
+      - *Location* - Approximate location of the provider
+      - *Storage System* - Offer is always connected to one Storage System like IPFS or Swarm
+      - *Offer Termination* - Provider can choose to discontinue his Offer. No new Agreements will be accepted, but current Agreements will be finished based on their current Billing Period.
+    - **Data Source** - Data Source is an abstract entity that defines the data that should be stored under an Agreement. There might be different mechanisms on how files should be retrieved, updated and removed.
+      - *IPFS Source* - Hash pointer to immutable data source.
+      - *Contract Source* - Pointer to different smart-contract that handles the mutability (eq. adding/removing files) 
+      - *IPNS Source* - [That is the future babe.](https://gph.is/1FD4aQ0)
+      - *RNS Source* - [That is also the future babe.](https://gph.is/1FD4aQ0)
+    - **Agreement** - Consumer creates Agreement for a specific Provider's Offer, specifying and the corresponding Data Source
+      - *Data Reference* - Pointer to Data Source
+      - *Agreement Termination* - Consumer can decide to terminate the agreement, which will take effect after finishing of current billing period.
+      - *Deposit Funds* - In order for an Agreement to be active, it needs to have available funds for each Billing Period renewal. Consumer can therefore deposit funds.
+      - *Payout Funds* - Upon finishing of a Billing Period, the Provider can request the funds for the finished periods to be transferred to him. 
```

Comment:
Implementation detail, but we also need the consumer to withdraw funds as they may make mistake and overfund

Reply:
Okey, maybe "Refund Funds"?

Reply:
Refund implies that the service was not satisfactory and the consumer gets the full amount out.

I think just *Withdraw  Deposit* could suffice

Reply:
Consumer can at all times withdraw (but not from the current billing period)

Reply:
Yeah but for that he has to cancel the Agreement right? He can't just take out money without termination right?
@vojtechsimetka is this something required?

Reply:
No he can always withdraw money until the amount that is locked which is the most imminent payment. I believe we already have this functionality

Reply:
Nope, currently only Provider can withdraw.

Reply:
Nope, currently only Provider can withdraw.

Reply:
I guess that is required then

Reply:
Yes I believe its not prepared to Withdraw as consumer unless you stop the Request. 
Lets add it here then we add it to the Contract.